portfolio/src/main/webapp/script.js

```diff
@@ -12,12 +12,111 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var context;
+var canvas;
+var myBubbleArray;
+var numBubs = 20;
+var animationID = null;
+// Resizes bubbles when page size is changed
+window.addEventListener('resize', startChat)
+
+function startChat() {
+  grabComments();
+  canvas = document.getElementById('myCanvas');
+  context = canvas.getContext('2d');
+  canvas.width = window.innerWidth;
+  canvas.height = window.innerHeight;
+  init();
+}
+
+class Bubble {
```

Comment:
script.js contains logic for both animation and the chat interface. Given that the two are functionally unrelated, you might want to separate them into two separate js files.

Reply:
On the other hand, a single JS file decreases load times.

In an ideal world, we would have two JS files + a build systems that bundles the two files together into one, but IMO this is fine. (Not to say you shouldn't implement Zach's suggestion; that's your choice.)